openmdao/solvers/ln_direct.py

```diff
@@ -60,9 +162,78 @@ def solve(self, rhs_mat, system, mode):
             for i in range(n_edge):
                 partials[:, i] = self.mult(ident[:, i])
 
-            deriv = np.linalg.solve(partials, rhs)
+        # Assemble the Jacobian
+        else:
 
-            self.system = None
-            sol_buf[voi] = deriv
+            # Must clear the jacobian if we switch modes
+            if self.mode != mode:
+                self.setup(system)
+            self.mode = mode
+            sys_name = system.name + '.'
+            conn = system.connections
 
-        return sol_buf
+            partials = self.jacobian
+            u_vec = system.unknowns
+
+            for sub in system.components(recurse=True, include_self=True):
```

Comment:
I'm leaving open the possibility that someone might want to attach this to a component. I think Justin does this.